src/sparsezoo/model/utils.py

```diff
@@ -562,29 +579,31 @@ def include_file_download_url(files: List[Dict]):
 
 
 def get_model_metadata_from_stub(stub: str) -> Dict[str, str]:
-    """
-    Return a dictionary of the model metadata from stub
-    """
-
-    stub_regex_expr = (
-        r"^(zoo:)?"
-        r"(?P<domain>[\.A-z0-9_]+)"
-        r"/(?P<sub_domain>[\.A-z0-9_]+)"
-        r"/(?P<architecture>[\.A-z0-9_]+)(-(?P<sub_architecture>[\.A-z0-9_]+))?"
-        r"/(?P<framework>[\.A-z0-9_]+)"
-        r"/(?P<repo>[\.A-z0-9_]+)"
-        r"/(?P<dataset>[\.A-z0-9_]+)"
-        r"/(?P<sparse_tag>[\.A-z0-9_-]+)"
-    )
-    matches = re.match(stub_regex_expr, stub)
-
-    return {
-        "domain": matches.group("domain"),
-        "sub_domain": matches.group("sub_domain"),
-        "architecture": matches.group("architecture"),
-        "sub_architecture": matches.group("sub_architecture"),
-        "framework": matches.group("framework"),
-        "repo": matches.group("repo"),
-        "dataset": matches.group("dataset"),
-        "sparse_tag": matches.group("sparse_tag"),
-    }
+    """Return a dictionary of the model metadata from stub"""
+
+    matches = re.match(STUB_V1_REGEX_EXPR, stub) or re.match(STUB_V2_REGEX_EXPR, stub)
+    if matches:
+        if "source_dataset" in matches.groupdict():
+            return {"repo_name": stub}
+
+        if "dataset" in matches.groupdict():
+            return {
+                "domain": matches.group("domain"),
+                "sub_domain": matches.group("sub_domain"),
+                "architecture": matches.group("architecture"),
+                "sub_architecture": matches.group("sub_architecture"),
+                "framework": matches.group("framework"),
+                "repo": matches.group("repo"),
+                "dataset": matches.group("dataset"),
+                "sparse_tag": matches.group("sparse_tag"),
+            }
+
+    return {}
```

Comment:
I'm tempted to untangle the handling of the two different formats. I might propose handling the new format separately from the old format:
```suggestion
    if re.match(STUB_V2_REGEX_EXPR, stub):
        return {"repo_name": stub}
    
    matches = re.match(STUB_V1_REGEX_EXPR, stub)
    if matches:
        return {
            "domain": matches.group("domain"),
            "sub_domain": matches.group("sub_domain"),
            "architecture": matches.group("architecture"),
            "sub_architecture": matches.group("sub_architecture"),
            "framework": matches.group("framework"),
            "repo": matches.group("repo"),
            "dataset": matches.group("dataset"),
            "sparse_tag": matches.group("sparse_tag"),
        }

    return {}
```

If you were to take this approach, you might not need any capture groups in `STUB_V2_REGEX_EXPR`

Reply:
Also, I don't know if it's important, but I usually think of `training_scheme` as a stub component. Doesn't seem like it really comes up in the data though.

Reply:
Re: `training_scheme`, not sure if there's a test for extracting metadata... but you might consider a test case that ensures that `zoo:cv/classification/resnet_v1-50/pytorch/sparseml/imagenet-augmented/pruned_quant-aggressive` is handled correctly and yields distinct metadata from `zoo:cv/classification/resnet_v1-50/pytorch/sparseml/imagenet/pruned_quant-aggressive`